examples/run_language_modeling.py

```diff
@@ -144,8 +145,88 @@ def __getitem__(self, i):
         return torch.tensor(self.examples[i], dtype=torch.long)
 
 
+class LazyLineByLineTextDataset(Dataset):
+    """
+    Credit: @bramvanroy for this linecache implementation.
+    """
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.num_entries = self._get_n_lines(self.file_path)
+
+    @staticmethod
+    def _get_n_lines(fin, size=65536):
+        # borrowed from https://stackoverflow.com/a/9631635/1150683
+        def blocks(files):
+            while True:
+                b = files.read(size)
+                if not b:
+                    break
+                yield b
+
+        with open(fin, encoding="utf-8") as fhin:
+            n_lines = sum(bl.count("\n") for bl in blocks(fhin))
+        return n_lines
+
+    def __getitem__(self, idx):
+        """
+
+        :param idx (int): the index of the line to get
+        :return (str or None): The line as a string (newline removed) or None if there is an exception.
+        """
+        # linecache starts counting from one, not zero, +1 the given index
+        return linecache.getline(self.file_path, idx + 1).rstrip()
+
+    def __len__(self):
+        return self.num_entries
+
+
+def make_collate(tokenizer, block_size, lazy=False):
+    """
+    Constructs collate function for Dataloader.
+
+    :param tokenizer: tokenizer object (instantiated) used to tokenize lines of text.
+    :param block_size: (int) tokenized sequences truncated so as to not exceed this length.
+    :param lazy: (bool) whether or not to use return a collate function for lazy text loading.
+    :return: (fn: list[str or torch.tensor[int] (1d)] -> torch.tensor[int] (2d)
+    """
+
+    def lazy_collate(examples):
+        """
+        Dataloader collate_fn for Lazy data loading.
+
+        :param examples: (list[str]) the text lines to be collated in this batch.
+        :return: (torch.tensor) of the tokenized examples padded/truncated to form a 2d tensor of ints.
+        """
+        # Filter empty strings. LazyLineByLineTextDataset will return empty string if there is an error on a line.
+        examples = [ex for ex in examples if ex]
+        examples = tokenizer.batch_encode_plus(
+            examples, max_length=block_size, return_tensors="pt", pad_to_max_length=True
+        )
```

Comment:
Tiny suggestion: try to avoid intermediate variables when it doesn't add much to readability. I'd rewrite as

```python
examples = tokenizer.batch_encode_plus(
            [ex for ex in examples if ex],
            max_length=block_size, return_tensors="pt", pad_to_max_length=True
)
```